types/block.go

```diff
@@ -677,10 +677,10 @@ type Commit struct {
 	// ValidatorSet order.
 	// Any peer with a block can gossip signatures by index with a peer without
 	// recalculating the active ValidatorSet.
-	Height     int64       `json:"height"`
-	Round      int         `json:"round"`
-	BlockID    BlockID     `json:"block_id"`
-	Signatures []CommitSig `json:"signatures"`
+	Height     int64         `json:"height"`
+	Round      int           `json:"round"`
+	BlockID    BlockID       `json:"block_id"`
+	Signatures [][]CommitSig `json:"signatures"`
```

Comment:
Modification to allow storing of multiple CommitSigs for each validator, which is required if validators send multiple CommitSigs with conflicting timestamps